detection_rules/rule.py

```diff
@@ -462,6 +460,61 @@ class ThresholdCardinality:
     threshold: ThresholdMapping
 
 
+@dataclass(frozen=True)
+class NewTermsRuleData(QueryRuleData):
+    """Specific fields for new terms field rule."""
+
+    @dataclass(frozen=True)
+    class NewTermsMapping(MarshmallowDataclassMixin):
+        @dataclass(frozen=True)
+        class HistoryWindowStart:
+            field: definitions.NonEmptyStr
+            value: definitions.NonEmptyStr
+
+        field: definitions.NonEmptyStr
+        value: definitions.NewTermsFields
+        history_window_start: List[HistoryWindowStart]
+
+    type: Literal["new_terms"]
+    new_terms: NewTermsMapping
+
+    def validate(self, meta: RuleMeta) -> None:
+        """Validates terms in new_terms_fields are valid ECS schema."""
+
+        super(NewTermsRuleData, self).validate_query(meta)
+
+        # validate history window start field exists
+        assert self.new_terms.history_window_start, \
+            "new terms field found with no history_window_start field defined"
+
+        # ecs validation
+        stack_version = Version(meta.get("min_stack_version",
+                                Version(Version(load_current_package_version()) + (0,))))
+        assert stack_version >= Version('8.4.0'), "New Terms rule types only compatible with 8.4.0+"
+        ecs_version = get_stack_schemas()[str(stack_version)]['ecs']
+        ecs_schema = ecs.get_schema(ecs_version)
+        for new_terms_field in self.new_terms.value:
+            assert new_terms_field in ecs_schema.keys(), \
+                f"{new_terms_field} not found in ECS schema (version {ecs_version})"
+
+        # validates length of new_terms to stack version - https://github.com/elastic/kibana/issues/142862
+        if stack_version >= Version('8.4.0') and stack_version < Version('8.6.0'):
+            assert len(self.new_terms.value) == 1, \
+                "new terms have a max limit of 1 for stack versions below 8.6.0"
```

Comment:
Nit: may be good to move '8.4.0' and '8.6.0' to constant variables.

e.g. `feature_min_stack = '8.4.0'`

Reply:
Added this inside `validate` method. I thought about defining it in the class, but that would result in it being defined in the final JSON object.